ktor-io/jvm/src/io/ktor/utils/io/internal/EventLoopExperimental.kt

```diff
@@ -1,8 +1,13 @@
+/*
+ * Copyright 2014-2020 JetBrains s.r.o and contributors. Use of this source code is governed by the Apache 2.0 license.
+ */
+
 package io.ktor.utils.io.internal
 
 import java.lang.reflect.*
 
-
+@Deprecated("Invoke processNextEventInCurrentThread() directly instead.", level = DeprecationLevel.ERROR)
```

Comment:
why deprecating internal class and removing it?

Reply:
We don't need it anymore. It was introduced to be compatible with different kotlinx.coroutines and fallback to some old behaviour. Nowadays, we don't need to support old coroutines so there is no need to do that magic anymore.

Reply:
but can we just delete this code then?

Reply:
sorry, my original comment had a typo, it should be
> why deprecating internal class and **not** removing it?

Reply:
Ah yes, you are right. Deleted.